contracts/extensions/ERC721AQueryable.sol

```diff
@@ -89,56 +111,76 @@ abstract contract ERC721AQueryable is ERC721A, IERC721AQueryable {
     ) external view virtual override returns (uint256[] memory) {
         unchecked {
             if (start >= stop) revert InvalidQueryRange();
-            uint256 tokenIdsIdx;
-            uint256 stopLimit = _nextTokenId();
             // Set `start = max(start, _startTokenId())`.
             if (start < _startTokenId()) {
                 start = _startTokenId();
             }
+            uint256 stopLimit = _nextTokenId();
             // Set `stop = min(stop, stopLimit)`.
-            if (stop > stopLimit) {
+            if (stop >= stopLimit) {
```

Comment:
Tbh, this `>=` is so that the compiler doesn't add an `iszero` behind the scenes. Functionally equivalent to `>`.